p2pd/main.go

```diff
@@ -17,8 +17,31 @@ import (
 	identify "github.com/libp2p/go-libp2p/p2p/protocol/identify"
 	multiaddr "github.com/multiformats/go-multiaddr"
 	promhttp "github.com/prometheus/client_golang/prometheus/promhttp"
+
+	_ "net/http/pprof"
 )
 
+func init() {
+	go func() {
+		for i := 6060; i < 7080; i++ {
+			addr := fmt.Sprintf("localhost:%d", i)
+			fmt.Printf("registering pprof debug http handler: %s\n", addr)
+			switch err := http.ListenAndServe(addr, nil); err {
```

Comment:
You can use `http.ListenAndServe(":0", nil)` to use free port but I think you want it to start with predictable number.

Reply:
Yeah, it's better UX to start with a predictable port.

Reply:
Yeah, that’s the goal.